client/my-sites/checklist/wpcom-checklist/index.jsx

```diff
@@ -524,23 +525,29 @@ function getContactPage( posts ) {
 	);
 }
 
-export default connect(
-	state => {
-		const siteId = getSelectedSiteId( state );
-		const siteSlug = getSiteSlug( state, siteId );
-
+const getTaskUrls = createSelector(
+	( state, siteId ) => {
 		const posts = getPostsForQuery( state, siteId, query );
 
 		const firstPost = find( posts, { type: 'post' } );
+		const siteSlug = getSiteSlug( state, siteId );
 		const contactPageID = get( getContactPage( posts ), 'ID', null );
 		const contactPageUrl = contactPageID && `/page/${ siteSlug }/${ contactPageID }`;
 
-		const user = getCurrentUser( state );
-
-		const taskUrls = {
+		return {
 			post_published: compact( [ '/post', siteSlug, get( firstPost, [ 'ID' ] ) ] ).join( '/' ),
```

Comment:
I know it's existing unmodified code, but the published post URL only makes sense when both `siteSlug` and `firstPost.ID` are valid.
```js
siteSlug && firstPost ? `/post/${ siteSlug }/${ firstPost.ID }` : null
```
looks better to me.

Reply:
Done in #28678.